src/block_queue.rs

```diff
@@ -21,6 +21,7 @@ use std::collections::{HashMap, VecDeque};
      engine::{Block, BlockId},
  };
 
+#[derive(Debug)]
```

Comment:
I think this is unrelated to the rest of the commit. If so, it should be in a separate commit.